plugins/dialog/plugin.js

```diff
@@ -500,6 +503,150 @@ CKEDITOR.DIALOG_STATE_BUSY = 2;
 
 		this.changeFocus = changeFocus;
 
+		function getCurrentIndex( startIndex, currentIndex, offset ) {
+			if ( startIndex < 0 ) {
+				startIndex = 0;
+			}
+
+			var dialogUiElements = me._.focusList,
+				direction = offset === 1 ? 'next' : 'prev',
+				isPreviousNodeInRadioGroup =
+					dialogUiElements[ startIndex ].role &&
+					dialogUiElements[ startIndex ].role === 'radiogroup',
+				isCurrentNodeInRadioGroup =
+					dialogUiElements[ currentIndex ].role &&
+					dialogUiElements[ currentIndex ].role === 'radiogroup';
+
+			// If the previous and the current element is not in the radio group or if the current
+			// element pointing outside the radio group the return the current index.
+			if (
+				( !isCurrentNodeInRadioGroup && !isPreviousNodeInRadioGroup ) ||
+				( !isCurrentNodeInRadioGroup && isPreviousNodeInRadioGroup )
+			) {
+				return currentIndex;
+			}
```

Comment:
I may be wrong, but can't be that simplified to:
```suggestion
			// If the previous and the current element is not in the radio group or if the current
			// element pointing outside the radio group the return the current index.
			if ( !isCurrentNodeInRadioGroup ) {
				return currentIndex;
			}
```
?